File: app/model/core/result/SuccessType.java
package model.core.result;

public enum SuccessType {

    FAILURE(0, "danger"), NONE(0, "danger"), PARTIALLY(1, "warning"), COMPLETE(2, "success");

    private int points;
    private String color;

    private SuccessType(int thePoints, String theColor) {
        points = thePoints;
        color = theColor;
    }

    public String getColor() {
        return color;
    }

    public int getPoints() {
        return points;
    }

}

File: app/model/core/result/SuccessType.scala
package model.core.result

import enumeratum.{EnumEntry, PlayEnum}

sealed abstract class SuccessType extends EnumEntry

object SuccessType extends PlayEnum[SuccessType] {

  override val values: IndexedSeq[SuccessType] = findValues

  case object ERROR extends SuccessType

  case object NONE extends SuccessType

  case object PARTIALLY extends SuccessType

  case object COMPLETE extends SuccessType

  def ofBool(success: Boolean): SuccessType = if (success) COMPLETE else NONE

}
